wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift

```diff
@@ -271,20 +271,36 @@ public class ZMUserSession: NSObject {
         let acmeApi = AcmeAPI(acmeDirectory: acmeDirectory)
         let httpClient = HttpClientImpl(
             transportSession: transportSession,
-            queue: syncContext)
+            queue: syncContext
+        )
+
         let apiProvider = APIProvider(httpClient: httpClient)
+
         syncContext.performAndWait {
             guard let coreCrypto = syncContext.coreCrypto else {
                 return
             }
+
             let e2eiClient = E2eIClient(coreCrypto: coreCrypto)
-            e2eiRepository = E2eIRepository(acmeApi: acmeApi,
-                                            apiProvider: apiProvider,
-                                            e2eiClient: e2eiClient)
+
+            let keyRotator = E2eIKeyPackageRotator(
+                coreCryptoProvider: coreCryptoProvider,
+                conversationEventProcessor: ConversationEventProcessor(context: syncContext),
```

Comment:
I think we don't need to create a new `ConversationEventProcessor`, ZMUserSession has `conversationEventProcessor` property